v-next/hardhat/src/internal/builtin-plugins/solidity/index.ts

```diff
@@ -13,6 +13,13 @@ const hardhatPlugin: HardhatPlugin = {
       );
       return artifactsPlugin;
     },
+    async () => {
+      // NOTE: builtin:solidity depends on buitin:coverage because it needs to
```

Comment:
This may lead to a circular dependency? I'd say we proceed with this as is, and then we'll see.